ext/nnstreamer/tensor_decoder/tensordec-imagesegment.c

```diff
@@ -0,0 +1,350 @@
+    idata->o_height = dim[1];
+    return TRUE;
+  } else if (op_num == 2) {
+    /* option3 = input model size (width:height) */
+    tensor_dim dim;
+
+    int rank = gst_tensor_parse_dimension (param, dim);
+
+    idata->i_width = 0;
+    idata->i_height = 0;
+    if (param == NULL || *param == '\0')
+      return TRUE;
+
+    if (rank < 2) {
+      GST_ERROR
+          ("mode-option-3 of image segmentation is input video dimension (WIDTH:HEIGHT). "
+          "The given parameter, \"%s\", is not acceptable.", param);
+      return TRUE;              /* Ignore this param */
+    }
+    if (rank > 2) {
+      GST_WARNING
+          ("mode-option-3 of image segmentation is input video dimension (WIDTH:HEIGHT). "
+          "The third and later elements of the given parameter, \"%s\", are ignored.",
+          param);
+    }
+    idata->i_width = dim[0];
+    idata->i_height = dim[1];
+    return TRUE;
+  }
+
+  GST_WARNING ("mode-option-\"%d\" is not definded.", op_num);
+  return TRUE;
+}
+
+/** 
+ * @brief tensordec-plugin's GstTensorDecoderDef callback 
+ * 
+ * [DeeplabV3 model]
+ * Just one tensor with [21(#labels):width:height:1], float32
+ * Probability that each pixel is assumed to be labeled object.
+ */
+static GstCaps *
+is_getOutCaps (void **pdata, const GstTensorsConfig * config)
+{
+  image_segments *idata = *pdata;
+  gint fn, fd;
+  GstCaps *caps;
+  char *str;
+
+  if (idata->mode == TFLITE_DEEPLAB_IMAGE_SEGMENT) {
+    g_return_val_if_fail (config != NULL, NULL);
+    GST_INFO ("Num Tensors = %d", config->info.num_tensors);
+    g_return_val_if_fail (config->info.num_tensors >= 1, NULL);
+  }
+
+  str = g_strdup_printf ("video/x-raw, format = RGBA, "
+      "width = %u, height = %u", idata->o_width, idata->o_height);
+  caps = gst_caps_from_string (str);
+  fn = config->rate_n; /** @todo Verify if this rate is ok */
+  fd = config->rate_d;
+
+  if (fn >= 0 && fd > 0) {
+    gst_caps_set_simple (caps, "framerate", GST_TYPE_FRACTION, fn, fd, NULL);
+  }
+  g_free (str);
+
+  return gst_caps_simplify (caps);
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static size_t
+is_getTransformSize (void **pdata, const GstTensorsConfig * config,
+    GstCaps * caps, size_t size, GstCaps * othercaps, GstPadDirection direction)
+{
+  return 0;
+  /** @todo Use appropriate values */
+}
+
+
+/** @brief Set color  according to each pixel's max probability  */
+static void
+set_color_according_to_label (image_segments * idata, GstMapInfo * out_info)
+{
+
+}
+
+
+/** @brief Set label_idx according to each pixel's label probabilities */
+static void
+set_label_index (image_segments * idata, void *data)
+{
+
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static GstFlowReturn
+is_decode (void **pdata, const GstTensorsConfig * config,
+    const GstTensorMemory * input, GstBuffer * outbuf)
+{
+  image_segments *idata = *pdata;
+  const size_t size = idata->i_width * idata->i_height * 4;
```

Comment:
eliminate hard-coded 4. use pre-defined one.

Reply:
Yes, I'll change it